src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModule.cs

```diff
@@ -43,6 +47,53 @@ public DiagnosticsTelemetryModule()
         }
 
         /// <summary>
+        /// Gets or sets a value indicating whether or not the Health Heartbeat feature is disabled.
+        /// </summary>
+        public bool IsHeartbeatEnabled
+        {
+            get
+            {
+                return this.HeartbeatProvider.IsEnabled;
+            }
+
+            set
+            {
+                this.HeartbeatProvider.IsEnabled = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the delay between heartbeats.
+        /// </summary>
+        public TimeSpan HeartbeatInterval
+        {
+            get => this.HeartbeatProvider.Interval;
+            set => this.HeartbeatProvider.Interval = value;
+        }
+
+        /// <summary>
+        /// Gets or sets property names that are not to be sent with the health heartbeats. null/empty list means allow all default properties through.
+        /// 
+        /// <remarks>
+        /// Default properties supplied by the Application Insights SDK:
+        /// - runtimeFramework
+        /// - baseSdkTargetFramework
+        /// </remarks>
+        /// </summary>
+        public IList<string> ExcludedHeartbeatProperties
+        {
+            get
+            {
+                return this.HeartbeatProvider.DisabledDefaultFields.ToList();
+            }
+
+            set
+            {
+                this.HeartbeatProvider.DisabledDefaultFields = value;
```

Comment:
why do you need a setter? Why not return editable `IList` and avoid potential null refs and executing external code?

Reply:
With the prior idea/design, we needed a way of knowing when to update the default fields (we could call ‘SetDefaultProperties’ with each pulse, but I think caching the values once is best). See below for reasoning and an update/correction to this design flaw.